components/infobox/wikis/apexlegends/infobox_weapon_custom.lua

```diff
@@ -0,0 +1,167 @@
+---
+-- @Liquipedia
+-- wiki=apexlegends
+-- page=Module:Infobox/Weapon/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Weapon = require('Module:Infobox/Weapon')
+local Class = require('Module:Class')
+local String = require('Module:StringUtils')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local CustomWeapon = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _weapon
+local _args
+
+local MAGAZINE_INFO = {
+	{},
+	{{text = '[[with Common Ext. Mag|<span class="white-text">EXT</span>]]', bgClass = 'gray-theme-dark-bg'}},
+	{{text = '[[with Rare Ext. Mag|<span class="white-text">EXT</span>]]', bgClass = 'sapphire-a2'}},
+	{
+		{text = '[[with Epic Ext. Mag|<span class="white-text">EXT</span>]]', bgClass = 'vivid-violet-theme-dark-bg'},
+		{text = '[[with Legendary Ext. Mag|<span class="black-text">EXT</span>]]', bgClass = 'bright-sun-0'}
+	},
+}
+local DAMAGE_INFO = {
+	{},
+	{{text = 'Head', bgClass = 'gray-theme-dark-bg', textBgClass = 'white-text'}},
+	{{text = 'Leg', bgClass = 'gray-theme-light-bg', textBgClass = 'black-text'}},
+}
+local NON_BREAKING_SPACE = '&nbsp;'
+
+function CustomWeapon.run(frame)
+	local weapon = Weapon(frame)
+	_weapon = weapon
+	_args = _weapon.args
+	weapon.addToLpdb = CustomWeapon.addToLpdb
+	weapon.createWidgetInjector = CustomWeapon.createWidgetInjector
+	return weapon:createInfobox(frame)
+end
+
+function CustomWeapon:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
```

Comment:
This function needs a lot of blank lines

Reply:
Do you mean blank lines separating each "table.insert(widgets, Cell{}" blocks etc?

Reply:
think so, it is to make the code better readable

Reply:
Added and committed